econsa/shapley.py

```diff
@@ -1,9 +1,214 @@
-"""Capabilities for computation of Shapley values.
-
-This module contains all we need to compute the Shapley values.
+"""Capabilities for computation of Shapley effects.
+This module contains functions to estimate shapley effects for models with
+dependent inputs.
 
 """
 
+import itertools
+import numpy as np
+import pandas as pd
+import chaospy as cp
+from econsa.sampling import cond_mvn
+
+
+def get_shapley(
+    method, model, Xall, Xcond, n_perms, n_inputs, n_output, n_outer, n_inner,
+):
+    """
+
+    This function calculates Shapley effects and their standard errors for
+    models with both dependent and independent inputs. We allow for two ways
+    to calculate Shapley effects: by examining all permutations of the given
+    inputs or alternatively, by randomly sampling permutations of inputs.
+
+    The function is a translation of the exact and random permutation funtions
+    in the ``sensitivity`` package in R, and takes the method (exact or random)
+    as an argument and therefore estimates shapley effects in both ways.
+
+
+    Parameters
+    ----------
+    method : string
+           Specifies which method you want to use to estimate shapley effects,
+           the ``exact`` or ``random`` permutations method. When the number of
+           inputs is small, it is better to use the ``exact`` method, and
+           ``random`` otherwise.
+
+    model : string
+          The model/function you will calculate the shapley effects on.
+
+    Xall : string (n)
+         A function that takes `n´ as an argument and generates a n-sample of
+         a d-dimensional input vector.
+
+    Xset : string (n, Sj, Sjc, xjc)
```

Comment:
Per [PEP](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), function or variable name should contain no upper case letter. Consider using format like `x_all`, etc.

Reply:
done

Reply:
done